### PR TITLE
Gpu picking

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/ShadersWGSL/picking.fragment.fx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+uniform meshID: vec4f;
+#endif
+
+@fragment
+fn main(input: FragmentInputs) -> FragmentOutputs {
+
+#if defined(INSTANCES)
+    fragmentOutputs.color = vMeshID;
+#else
+	fragmentOutputs.color = meshID;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/picking.fragment.fx:14]:
```suggestion
	fragmentOutputs.color = uniforms.meshID;
```

File: packages/dev/core/src/ShadersWGSL/picking.fragment.fx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+varying vMeshID: vec4f;
+#else
+uniform meshID: vec4f;
+#endif
+
+@fragment
+fn main(input: FragmentInputs) -> FragmentOutputs {
+
+#if defined(INSTANCES)
+    fragmentOutputs.color = vMeshID;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/picking.fragment.fx:12]:
```suggestion
    fragmentOutputs.color = input.vMeshID;
```

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15441/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15441/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15441/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15441/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15441/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15441/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15441/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15441/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15441/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15441/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15441/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15441/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Pull request merged**